include/solvers/vof_phase_gradient_projection.h

```diff
@@ -0,0 +1,145 @@
+// SPDX-FileCopyrightText: Copyright (c) 2020-2024 The Lethe Authors
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception OR LGPL-2.1-or-later
+
+#ifndef lethe_vof_phase_gradient_projection_h
+#define lethe_vof_phase_gradient_projection_h
+
+#include <solvers/vof_linear_subequations_solver.h>
+
+/**
+ * @brief VOF phase fraction gradient L2 projection solver.
+ *
+ * @tparam dim Number of dimensions of the problem.
+ *
+ * @tparam ScratchDataType Type of scratch data object used for linear system
+ * assembly.
+ */
+template <int dim, typename ScratchDataType>
+class VOFPhaseGradientProjection
+  : public VOFLinearSubequationsSolver<dim, ScratchDataType>
+{
+public:
+  /**
+   * @brief Constructor for the L2 projection of the VOF phase fraction gradient
+   * (pfg)
+   *   *
+   * @param[in,out] p_subequations Subequations interface object used to get
+   * information from other subequations and store information from the
+   current
+   * one.
+   *
+   * @param[in] p_multiphysics Multiphysics interface object used to get
+   * information from physics.
+   *
+   * @param[in] p_simulation_parameters Simulation parameters.
+   *
+   * @param[in] p_triangulation Distributed mesh information.
+   *
+   * @param[in] p_simulation_control Object responsible for the control of
+   * steady-state and transient simulations. Contains all the information
+   * related to time stepping and the stopping criteria.
+   *
+   * @param[in] p_pcout Parallel cout used to print the information.
+   */
+  VOFPhaseGradientProjection(
+    VOFSubequationsInterface<dim>   *p_subequations,
+    MultiphysicsInterface<dim>      *p_multiphysics,
+    const SimulationParameters<dim> &p_simulation_parameters,
+    std::shared_ptr<parallel::DistributedTriangulationBase<dim>>
+                                       &p_triangulation,
+    std::shared_ptr<SimulationControl> &p_simulation_control,
+    const ConditionalOStream           &p_pcout)
+    : VOFLinearSubequationsSolver<dim, ScratchDataType>(
+        VOFSubequationsID::phase_gradient_projection,
+        p_subequations,
+        p_multiphysics,
+        p_simulation_parameters,
+        p_triangulation,
+        p_simulation_control,
+        p_simulation_parameters.multiphysics.vof_parameters
+          .surface_tension_force.verbosity,
+        p_pcout)
+  {}
+
+  /**
+   * @brief Default destructor.
+   */
+  ~VOFPhaseGradientProjection() = default;
+
+  /**
+   * @brief Set up the DofHandler and the degree of freedom associated with
+   * the physics.
+   *
+   * @param[in] subequation_id Identifier corresponding to the subequation,
+   * for terminal print.
+   */
+  void
+  setup_dofs() override
+  {
+    VOFLinearSubequationsSolver<dim, ScratchDataType>::setup_dofs();
+  }
```

Comment:
Why do you override it if you are calling the parent class' function?

Reply:
Nice catch, these are remaining traces from previous tests I was doing when generalizing it ^^'